scripts/apps/desks/directives/DesksReactDropdown.jsx

```diff
@@ -0,0 +1,79 @@
+/**
+ * @ngdoc directive
+ * @module superdesk.apps.desks
+ * @name DesksReactDropdown
+ *
+ * @requires React
+ * @requires item
+ * @requires className
+ * @requires desks
+ * @requires noDesksLabel
+ *
+ *
+ * @description Creates dropdown react element with list of available desks
+ */
+import React from 'react';
+
+DesksReactDropdown.$inject = ['item', 'className', 'desks', 'noDesksLabel'];
+export function DesksReactDropdown(item, className, desks, noDesksLabel) {
+    /** Dropdown component */
+    class MarkDesksDropdown extends React.Component {
```

Comment:
This component should be in a different file. You can pass angular injectables via props for now.